### DISCUSSION TITLE
Can I search by whole object?

### DISCUSSION THREAD

**Discussion by cadilhac** [ravendb/ravendb#19224] [Q&A]
Right now, my db stores a collection of objects with various fields (simple ints and strings). To query this collection, trying to match a specific set of properties, I have to write a Where clause with equalities for every possible property (and use a corresponding index).

`session.Query<>().Where(x => x.a == test.a && x.b == test.b && .....)`

Is there no way to just pass test to a query and it will test all its fields, returning the record that matches?

**Comment by ayende** [comment]
You can do that, yes, see:

https://ayende.com/blog/153729/lazys-man-comprehensive-search-with-ravendb

**Reply by cadilhac** [reply]
Hmm... Does not apply here I think. I don't want to compare all properties of entities against a single value. I want to compare entities to a complete object with all its properties. Actually, to do that, I could store a hash of all the properties inside the entities and query by the hash. But my question was if it could be done more directly inside RavenDb,

**Reply by ayende** [reply]
Oh, I see. 
There are a few ways of doing that, but hashing the value (sha2 etc) should be sufficient.

You _can_ do: `x=>x.nestedValue == otherValue` and that would compare the JSON *strings*.
But that can be really expensive for large objects.

Note that if you are using hashing, make sure that you're normalizing the data (no whitespace, sorted properties, etc).

**Reply by cadilhac** [reply]
Not sure about your example to compare json strings, but yeah, I will end up using a hash. MD5 maybe... for something short.
